server/sonar-server/src/test/java/org/sonar/server/computation/step/ComplexityMeasuresStepTest.java

```diff
@@ -0,0 +1,224 @@
+import static org.sonar.server.computation.measure.MeasureRepoEntry.entryOf;
+import static org.sonar.server.computation.measure.MeasureRepoEntry.toEntries;
+
+public class ComplexityMeasuresStepTest {
+
+  private static final int ROOT_REF = 1;
+  private static final int MODULE_REF = 11;
+  private static final int SUB_MODULE_REF = 111;
+  private static final int DIRECTORY_REF = 1111;
+  private static final int FILE_1_REF = 11111;
+  private static final int FILE_2_REF = 11121;
+
+  private static final DumbComponent MULTIPLE_FILES_TREE = builder(PROJECT, ROOT_REF)
+    .addChildren(
+      builder(MODULE, MODULE_REF)
+        .addChildren(
+          builder(MODULE, SUB_MODULE_REF)
+            .addChildren(
+              builder(DIRECTORY, DIRECTORY_REF)
+                .addChildren(
+                  builder(FILE, FILE_1_REF).build(),
+                  builder(FILE, FILE_2_REF).build()
+                ).build()
+            )
+            .build()
+        ).build()
+    ).build();
+
+  @Rule
+  public BatchReportReaderRule reportReader = new BatchReportReaderRule();
+
+  @Rule
+  public TreeRootHolderRule treeRootHolder = new TreeRootHolderRule().setRoot(MULTIPLE_FILES_TREE);
+
+  @Rule
+  public MetricRepositoryRule metricRepository = new MetricRepositoryRule()
+    .add(COMPLEXITY)
+    .add(COMPLEXITY_IN_CLASSES)
+    .add(COMPLEXITY_IN_FUNCTIONS)
+    .add(FUNCTION_COMPLEXITY_DISTRIBUTION)
+    .add(FILE_COMPLEXITY_DISTRIBUTION)
+    .add(CLASS_COMPLEXITY_DISTRIBUTION)
+    .add(FILE_COMPLEXITY)
+    .add(FILES)
+    .add(CLASS_COMPLEXITY)
+    .add(CLASSES)
+    .add(FUNCTION_COMPLEXITY)
+    .add(FUNCTIONS)
+    ;
+
+  @Rule
+  public MeasureRepositoryRule measureRepository = MeasureRepositoryRule.create(treeRootHolder, metricRepository);
+
+  private ComputationStep underTest = new ComplexityMeasuresStep(treeRootHolder, metricRepository, measureRepository);
+
+  @Test
+  public void aggregate_complexity() throws Exception {
+    verify_sum_aggregation(COMPLEXITY_KEY);
+  }
+
+  @Test
+  public void aggregate_complexity_in_classes() throws Exception {
+    verify_sum_aggregation(COMPLEXITY_IN_CLASSES_KEY);
+  }
+
+  @Test
+  public void aggregate_complexity_in_functions() throws Exception {
+    verify_sum_aggregation(COMPLEXITY_IN_FUNCTIONS_KEY);
+  }
+
+  private void verify_sum_aggregation(String metricKey){
+    measureRepository.addRawMeasure(FILE_1_REF, metricKey, newMeasureBuilder().create(10));
+    measureRepository.addRawMeasure(FILE_2_REF, metricKey, newMeasureBuilder().create(40));
+
+    underTest.execute();
+
+    assertThat(measureRepository.getAddedRawMeasure(FILE_1_REF, metricKey)).isAbsent();
+    assertThat(measureRepository.getAddedRawMeasure(FILE_2_REF, metricKey)).isAbsent();
+
+    int expectedNonFileValue = 50;
+    assertThat(toEntries(measureRepository.getAddedRawMeasures(DIRECTORY_REF))).contains(entryOf(metricKey, newMeasureBuilder().create(expectedNonFileValue)));
```

Comment:
No, because I only want to test to aggregation of one specific measure. As another measure is generated, it's not possible to use containsOnly. Otherwise it will test 2 different measures, and it's not what I want.